### PR TITLE
converting MesheryPlaycomponent into functional component 

### LINKED ISSUE #8801: [UI] Convert `MesheryPlayComponent` from class based to functional component
### Current Behavior
`MesheryPlayComponent` is a class-based component.

### Desired Behavior
`MesheryPlayComponent` is a functional component. 

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Cvr421**
**Notes for Reviewers**
converting MesheryPlaycomponent from Class based component  into functional component

This PR fixes #8801

Current Behaviour 
Class Based component 

Expected Behaviour 
Functional Based component 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
 Yes, I signed my commits.

### CODE REVIEW

**Comment by Cvr421** [comment]
@Rajdip019 @aabidsofi19  Please review this PR .

**Review by Rajdip019 — APPROVED** [review]

File: ui/components/MesheryPlayComponent.js [hunk 1]
```diff
@@ -75,54 +75,48 @@ const styles = (theme) => ({
...
     }
-  }
 
-  componentWillUnmount() {
-    this.props.router.events.off("routeChangeComplete", this.handleRouteChange);
-  }
+    return () => {
+      router.events.off("routeChangeComplete", handleRouteChange);
+    };
+  }, [router, meshAdapters]); // Dependencies for useEffect
```
Comment (Yashsharma1911) [inline ui/components/MesheryPlayComponent.js:113]: Thanks for adding comment but I think this comment is not required
```suggestion
  }, [router, meshAdapters]);
```

**Review by Yashsharma1911 — APPROVED** [review]

**Comment by saurabh100ni** [comment]
Let's discuss this on the Meshery dev call tomorrow,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Comment by Cvr421** [comment]
@aabidsofi19  Can you please review this PR . I think it is good to go . And it is opened from very long time .

**Comment by aabidsofi19** [comment]
@Mohith234  if you would review this  ..

**Comment by Cvr421** [comment]
@Mohith234 Can you please review this PR.

File: ui/components/MesheryPlayComponent.js [hunk 2]
```diff
@@ -75,98 +75,87 @@ const styles = (theme) => ({
   },
 });
 
-class MesheryPlayComponent extends React.Component {
-  constructor(props) {
-    super(props);
+function MesheryPlayComponent(props) {
```
Comment (Mohith234) [inline ui/components/MesheryPlayComponent.js:78]: - Can we use Arrow function here?
```suggestion
const MesheryPlayComponent = (props) => {
```

Comment (Mohith234) [reply ui/components/MesheryPlayComponent.js:78]: The rest looks good to me 🚀

**Comment by Cvr421** [comment]
@Mohith234  I have made the changes requested by you and Please review . I think we are good to go .

**Pull request closed without merging**